cmd/pinniped-proxy/src/service.rs

```diff
@@ -9,11 +10,34 @@ use crate::https;
 pub async fn proxy(req: Request<Body>) -> Result<Response<Body>, Infallible> {
     let log_data = logging::request_log_data(&req);
 
+    let _ = match https::get_api_server_url(req.headers()) {
```

Comment:
The `_` is just to avoid warnings about unused variables (I'll use the in later PRs), similar to golang (though go forces you rather than warns).